### ISSUE TITLE
Pools of Darkness Game (AD&D Gold Box) Premature Crash

### ISSUE THREAD

**Issue by B00TP** [joncampbell123/dosbox-x#2275]
**Describe the bug**
On launching START.BAT, the initial AD&D splash screen / introduction display correctly (VACE.EXE), but the game then terminates on invoking COMMAND.EXE / GAME.EXE. After testing further, I can confirm this issue does not occur in 0.82.20, but first occurs in 0.83.21.

START.BAT contains the following:

startup
if exist snd.com snd
control %1 %2 %3 %4
if exist snd.com snd U

**To Reproduce**
Steps to reproduce the behaviour:
1. Install Pools of Darkness game (v1.10 - the only version available).
2. Run START.BAT.
3. AD&D splash screen / introduction display correctly.
4. Game crashes and error is returned, stating 'Sound Driver unloaded'.

**Expected behaviour**
After clicking though or pressing the ESC key on the AD&D splash screen / introduction, COMMAND.EXE / GAME.EXE should launch.

**Environment (please complete the following information):**
 - Microsoft Windows 7 32-bit.
 - Using DOSBox-X 0.83.10, but first occurs in 0.83.21.
 - Tested with standard 'out-of-box' configuration.

**Additional context**
I cannot replicate the problem in other DOSBox releases. In debugging with a different SVN, I noticed the following messages at this time: 'Adlib polling hack triggered. Forcing timers to reset'. I have tried changing the option 'adlib force timer overflow on detect', but it did not fix the problem. Turning the sound off also made no difference.

**Comment by B00TP** [comment]
After investigating a little further, typing LOADFIX before running START.EXE resolves the issue. Seems to be OK but I wasn't sure if any other parameters were required for the LOADFIX command.

**Comment by Wengier** [comment]
I have tried the game. I think the said "crash" is a result of the following commit: https://github.com/joncampbell123/dosbox-x/commit/331cb7ab3ae51584a470dd9a67b575131c3ee1a0

If you use LOADFIX for this, you can just change ```control %1 %2 %3 %4``` to ```loadfix control %1 %2 %3 %4``` in start.bat and it should work. 

Another solution is to change  ```dos=high, umb``` to ```dos=low, umb``` in [config] section of the config file before running DOSBox-X.

**Comment by B00TP** [comment]
Thanks, I tried running GAME.EXE directly and it would seem Microsoft's EXEPACK algorithm is to blame. DOSBox-X launches GAME.EXE successfully without LOADFIX, but warns ‘Packed file is corrupt’. Testing further, if I disable autoa20fix then it just halts at ‘Packed file is corrupt’. However, the autoa20fix does not help when running START.BAT; I imagine due to the way the other .EXE files (via COMMAND.EXE) are invoked.

Regardless, running LOADFIX first to reside in the lowest 64KB works fine.

**Comment by Wengier** [comment]
Yes, it has turned out to be a "Packed file is corrupt" error. But in such case GAME.EXE is called by CONTROL.EXE (instead of directly or by START.BAT). autoa20fix cannot help with that since it won't come in when an executable program is called by another executable program. The commit mentioned above was the first commit (between 0.82.20 and 0.83.21) that the "crash" occurred (by doing git bisect), but it was mainly for finding clues/possible solutions and as you said Microsoft's EXEPACK algorithm is to be blamed.
